src/MatrixAlgebra.jl

```diff
@@ -69,6 +69,11 @@ is_zero_divisor(a::MatAlgElem{T}) where T <: RingElement = is_zero_divisor(det(a
 
 is_zero_divisor(a::MatAlgElem{T}) where T <: FieldElement = rank(a) != degree(a)
 
+function is_zero_divisor_with_annihilator(a::MatAlgElem{T}) where T <: RingElement
+   f, b = is_zero_divisor_with_annihilator(det(a))
+   throw(NotImplementedError(:adj, A)) #return f, b*adj(A)
```

Comment:
@thofma how hard would it be to whip up an `adj` implementation?

Reply:
Would be easy, if someone would have taken care of the sign in `pseudo_inverse`:

```

  pseudo_inv(M::MatrixElem{T}) where {T <: RingElement}

  Given a non-singular n\times n matrix M over a ring return a tuple X, d
  consisting of an n\times n matrix X and a denominator d such that MX = dI_n,
  where I_n is the n\times n identity matrix. The denominator will be the
  determinant of M up to sign. If M is singular an exception is raised.
```
Notice the "up to sign".

Reply:
up to sign isn't a problem, the prohibition against singular matrices is the problem.

Reply:
If you really want the adjugate, then the sign is a problem?